### PR TITLE
Gtk3 transition initial changes

### PR DESCRIPTION
**Pull request by cristina-suteu**
replaced deprecated widgets
replaced deprecated variable types
fixed plugins 

TO DO: 
update all glade files to match new widget types and hierarchy

### CODE REVIEW

File: plugins/AD7303.c [hunk 1]
```diff
@@ -54,18 +54,18 @@ static GtkWidget *radio_waveform;
 static GtkWidget *databox;
 static GtkWidget *preview_graph;
 
-static GdkColor color_background = {
+static GdkRGBA color_background = {
```
Comment (dNechita) [inline plugins/AD7303.c:57]: Should you also add the alpha channel? (besides red, green and blue)

File: plugins/block_diagram.h [hunk 2]
```diff
@@ -81,13 +81,15 @@ static gboolean draw_block_diagram(GtkWidget *widget, cairo_t *cr, GtkImage *blo
 	static double scale = 0;
 	int x_big, y_big, x_click, y_click, x_in, y_in;
 	char name[256];
+	GtkAllocation* allocation = g_new(GtkAllocation, 1);
+    gtk_widget_get_allocation(widget, allocation);
```
Comment (dNechita) [inline plugins/block_diagram.h:85]: Not aligned with the rest of code

Comment (dNechita) [inline plugins/block_diagram.h:85]: can't get_allocated_width() & get_allocated_height be used instead?

File: plugins/dac_data_manager.c [hunk 3]
```diff
@@ -2318,4 +2368,4 @@ unsigned dac_data_manager_dds_tone(unsigned tx_index,
 	enum dds_tone_index tone_index, enum dds_tone_type tone_type)
 {
 	return ((tx_index * TX_NB_TONES) + (tone_index * CHANNEL_NB_TONES) + tone_type);
-}
+}
```
Comment (dNechita) [inline plugins/dac_data_manager.c:2371]: missing newline at the end of file

File: plugins/dac_data_manager.c [hunk 4]
```diff
@@ -826,18 +826,16 @@ static GtkWidget *spin_button_create(double min, double max, double step, unsign
 static GtkWidget *frame_with_table_create(const char *frm_title, unsigned rows, unsigned columns)
 {
 	GtkWidget *frame;
-	GtkWidget *align;
+	// GtkWidget *box;
```
Comment (dNechita) [inline plugins/dac_data_manager.c:829]: What the plan with these commented lines? Can you leave a to-do next to the code?

File: CMakeLists.txt [hunk 5]
```diff
@@ -91,8 +91,8 @@ endif()
 
 set(OSC_SRC osc.c oscplot.c datatypes.c iio_widget.c iio_utils.c
 	fru.c dialogs.c trigger_dialog.c xml_utils.c libini/libini.c
-	libini2.c phone_home.c #plugins/dac_data_manager.c
```
Comment (dNechita) [inline CMakeLists.txt:94]: please keep the tabs

File: oscplot.c [hunk 6]
```diff
@@ -7030,8 +7018,18 @@ static void create_plot(OscPlot *plot)
...
 		TRUE, TRUE, TRUE, TRUE);
 	gtk_box_pack_start(GTK_BOX(priv->capture_graph), table, TRUE, TRUE, 0);
-	// TO DO: handle this using GTK3 (probably using stylesheets)
-	// gtk_widget_modify_bg(priv->databox, GTK_STATE_NORMAL, &color_background);
+
+	GtkCssProvider *provider;
+	GtkStyleContext *style_context;
+	GdkDisplay *display;
+	GdkScreen *screen;
+	provider = gtk_css_provider_new();
```
Comment (dNechita) [inline oscplot.c:7026]: Should the 'provider' be freed at some point?

**Comment by dNechita** [comment]
> Is this ready to be run time tested yet?

Not yet. We still need to transition all glade files from GTK2 to GTK3.

File: plugins/AD7303.c (hunk 1 above)
Comment (cristina-suteu) [reply plugins/AD7303.c:57]: added alpha channel for all structures

File: plugins/dac_data_manager.c (hunk 4 above)
Comment (cristina-suteu) [reply plugins/dac_data_manager.c:831]: added suggestive TO DO comments

File: oscplot.c (hunk 6 above)
Comment (cristina-suteu) [reply oscplot.c]: provider is now freed when plot object is destroyed

**Pull request merged**